service/server/main.py

```diff
@@ -0,0 +1,211 @@
+# even when `--check_ref` is used because the headers line is still produced.
+SAMPLE_VARIANT = 'chr1	807692	.	C	A	0	PASS	KM=11.5	GT	1|1'
+
+
+def setup_vep_cache():
+    """Initialization function to download and build VEP's cache"""
+    # TODO(jessime) These values should be loaded from ENV variables
+    local_cache = '/mnt/vep/vep_cache/'
+    gcs_dir = 'gs://gcp-variant-annotation-vep-cache'
+    vep_file = 'vep_cache_homo_sapiens_GRCh38_91.tar.gz'
+    gcs_vep_file = os.path.join(gcs_dir, vep_file)
+    local_vep_file = os.path.join(local_cache, vep_file)
+    check_call('gsutil cp {} {}'.format(gcs_vep_file, local_cache).split())
+    check_call('tar xzvf {} -C {}'.format(local_vep_file, local_cache).split())
+    # TODO(jessime) Here and elsewhere, replace prints with proper logging
+    print 'VEP CACHE setup.'
+
+
+#Global calls after `setup_vep_cache` is defined but before `app` decorators
+if os.environ['ANNOTATION_SERVER_ENV'] == 'production':
+    # TODO(jessime) setup configurations for local vs production environments
+    setup_vep_cache()
+app = Flask(__name__)
+
+
+def run_vep(variants):
+    """Execute VEP in a subprocess.
+
+    Args:
+        variants: A single str of vcf formatted variants to be passed to VEP
+    Returns:
+        response: Dict storing stdout and stderr from VEP subprocess run
+    """
+    # type: (str) -> Dict[str, str]
+    cmd = 'run_vep.sh'
+    p = Popen(['bash', cmd], stdout=PIPE, stdin=PIPE, stderr=PIPE)
+    stdout, stderr = p.communicate(input=variants)
+    # print('perl stdout: ', stdout)
+    # TODO (jessime) Figure out how to deal with harmless warnings.
+    # Also figure out how to deal with more important warnings like failing
+    # `--check_ref` because the assembly is incorrect
+    if stderr.startswith('Possible precedence'):
+        stderr = stderr[stderr.index('\n')+1:]
+    response = {'stdout': stdout, 'stderr': stderr}
+    return response
+
+
+def extract_headers_and_annotations(vep_tabbed):
+    """Separate the header line and annotation lines into seperate variables.
+
+    Args:
+        vep_tabbed: str of VEP output in tabbed format. For details see:
+            http://useast.ensembl.org/info/docs/tools/vep/vep_formats.html#tab
+    Returns:
+        header_names: List of annotation header names. e.g. ['GENE', 'IMPACT']
+        annotation_lines: All annotations provided by VEP
+    """
+    # type: (str) -> Tuple[List[str], List[str]]
+    header_names = None
+    vep_tabbed_lines = vep_tabbed.split('\n')[:-1]  # remove trailing empty line
+    annotation_lines = []
+    while vep_tabbed_lines:
+        line = vep_tabbed_lines.pop(0)
+        if line.startswith('#'):
+            header_names = line
+        else:
+            annotation_lines = [line] + vep_tabbed_lines
+            break
+    header_names = header_names.split('\t')
+    return header_names, annotation_lines
+
+
+def vep_tab_to_json(response):
```

Comment:
So it seems `response` is both input and output for this function, correct? If yes, can you separate it and do not mutate the input. Also it would be more readable that any `response` that is not over the network be renamed to something more descriptive, e.g., `annotation_response` or maybe `annotation_stdout`? I thought that should make understanding the code easier.